src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.h

```diff
@@ -0,0 +1,210 @@
+
+    /**
+     * @brief
+     *   Called when the server receives the ProvideAnswer command.
+     *
+     *   In the standard WebRTC Offer/Answer flow, once the provider sends an Offer (or
+     *   the requestor has re-offered), the other side will respond with an SDP Answer.
+     *   This method is invoked to store and process that Answer.
+     *
+     * @param[in] sessionId  The identifier of the current session.
+     * @param[in] sdpAnswer  The SDP Answer string.
+     *
+     * @return CHIP_ERROR
+     *   - CHIP_NO_ERROR if the Answer is accepted
+     *   - Appropriate error if invalid or unexpected
+     */
+    virtual CHIP_ERROR HandleProvideAnswer(uint16_t sessionId, const chip::CharSpan & sdpAnswer) = 0;
+
+    /**
+     * @brief
+     *   Called when the server receives the ProvideICECandidates command.
+     *
+     *   After the initial Offer/Answer exchange, additional ICE Candidates may be
+     *   discovered. This command allows the requestor to provide those candidates
+     *   (i.e., Trickle ICE).
+     *
+     * @param[in] sessionId  The current session ID.
+     * @param[in] candidates A list of ICE candidate strings.
+     *
+     * @return CHIP_ERROR
+     *   - CHIP_NO_ERROR on success
+     *   - An error if the session is invalid or the candidates cannot be processed
+     */
+    virtual CHIP_ERROR HandleProvideICECandidates(uint16_t sessionId, const std::vector<std::string> & candidates) = 0;
+
+    /**
+     * @brief
+     *   Called when the server or the requestor ends a session via the EndSession command.
+     *
+     * @param[in] sessionId  The ID of the session to end.
+     * @param[in] reasonCode Reason for ending the session (e.g. normal closure, resource limit).
+     *
+     * @return CHIP_ERROR
+     *   - CHIP_NO_ERROR on success
+     *   - Error if no matching session is found or some cleanup error occurs
+     */
+    virtual CHIP_ERROR HandleEndSession(uint16_t sessionId, WebRTCEndReasonEnum reasonCode) = 0;
+};
+
+class WebRTCTransportProviderServer : public AttributeAccessInterface, public CommandHandlerInterface
+{
+public:
+    /**
+     * @brief
+     *   Constructs the WebRTCTransportProviderServer with the specified delegate and endpoint.
+     *
+     * @param[in] delegate   A pointer to an implementation of the Delegate interface. Must remain
+     *                       valid for the lifetime of this object.
+     * @param[in] endpointId The Endpoint where the WebRTC Transport Provider cluster is published.
+     */
+    WebRTCTransportProviderServer(Delegate * delegate, EndpointId endpointId);
+
+    /**
+     * @brief
+     *   Destructor. Cleans up any internal data, but does not destroy the delegate.
+     */
+    ~WebRTCTransportProviderServer() override;
+
+    /**
+     * @brief
+     *   Initializes this server instance.
+     *
+     * @return CHIP_ERROR
+     *   - CHIP_NO_ERROR on successful registration
+     *   - Error code if the registration failed
```

Comment:
this does not specify what "registration". I would consider totally cleaning this up.
Maybe a comment as to why there is no `Shutdown` would help ... it seems the destructor is the shutdown. I would consider changing it to have shutdown and destructor calling shutdown.

Also see previous comments about `return CHIP_ERROR` often being obvious.